src/helper.rs

```diff
@@ -1,21 +1,21 @@
 //! Defines some things used from different modules but not to be exported.
 
-use reqwest::header::Headers;
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
 use std::collections::BTreeMap;
 use std::iter::FromIterator;
 
-pub fn serialize_headers(headers: &Headers) -> BTreeMap<String, String> {
+pub fn serialize_headers(headers: &HeaderMap) -> BTreeMap<String, String> {
     let tuples_iter = headers
         .iter()
-        .map(|hv| (hv.name().to_string(), hv.value_string()));
+        .map(|(hn, hv)| (hn.to_string(), hv.to_str().unwrap().to_string()));
 
     BTreeMap::<String, String>::from_iter(tuples_iter)
 }
 
-pub fn deserialize_headers(map: &BTreeMap<String, String>) -> Headers {
-    let mut headers = ::reqwest::header::Headers::new();
+pub fn deserialize_headers(map: &BTreeMap<String, String>) -> HeaderMap {
+    let mut headers = ::reqwest::header::HeaderMap::new();
     for (name, value) in map.iter() {
-        headers.append_raw(name.clone(), value.as_bytes().to_vec())
+        headers.insert(HeaderName::from_bytes(&name.clone().into_bytes()).unwrap(), HeaderValue::from_str(value).unwrap());
```

Comment:
Nitpick: `HeaderName::from_bytes(&name.as_ref()).unwrap()` avoids a clone. (Might have been optimized away by the compiler anyway.)